### PR TITLE
Fix trailing comma issue

### PR DESCRIPTION
**Pull request by jeremy-rifkin**
Make trailing commas more consistent throughout the project, fixes config conflict between eslint and prettier. Resolves an oversight in #4766.

### CODE REVIEW

File: .eslintrc.yml [hunk 1]
```diff
@@ -32,7 +32,7 @@ rules:
       objects: always-multiline
       imports: always-multiline
       exports: always-multiline
-      functions: always-multiline
+      functions: never
```
Comment (jeremy-rifkin) [inline .eslintrc.yml:35]: Bringing root .eslintrc in-line with static's eslintrc

File: .prettierrc.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 module.exports = {
-    trailingComma: 'all',
+    trailingComma: 'es5',
```
Comment (jeremy-rifkin) [inline .prettierrc.js:2]: Bringing the base prettierrc in-line with static

**Review by mattgodbolt — APPROVED** [review]
I'm used to python's `black` which forces them on the end (making it easier to add new lines and/or reorder list params. I'm ok with this though - but any reason for the change?

File: .prettierrc.js (hunk 2 above)
Comment (matsjla) [reply .prettierrc.js:2]: I personally think trailing commas make for easier to read diffs (as you don't have a diff on the line with the new, added comma)

**Review by matsjla — CHANGES_REQUESTED** [review]
I'm not sure why we have any eslint rules for formatting, as prettier should just take care of everything. Besides, the prettier plugin for eslint disables any eslint rules that would conflict with the prettier config (making prettier the source of truth)

I thus propose we instead remove the formatting rules from the eslint config

**Comment by jeremy-rifkin** [comment]
@mattgodbolt @matsjla Thanks for taking a look, at the moment static doesn't use trailing commas for function parameters but the backend does. Happy to instead update static to be in-line with the backend.

**Comment by mattgodbolt** [comment]
@jeremy-rifkin Would prefer that but...also just grateful to have someone looking in to these things :) thank you!

File: lib/.eslint-ce-lib.yml [hunk 3]
```diff
@@ -17,17 +18,17 @@ rules:
...
     - error
     - always
   eqeqeq:
     - error
     - smart
   indent:
-    - error
-    - 4
-    - SwitchCase: 1
+    - off
```
Comment (jeremy-rifkin) [inline lib/.eslint-ce-lib.yml:29]: There's an edge case or two eslint and prettier disagree on for indentation. Two files in lib/ are affected despite specifying the prettier eslint plugin. Just turning indent off for now.

File: lib/handlers/formatting.ts [hunk 4]
```diff
@@ -64,8 +64,8 @@ export class FormattingHandler {
             const version = hasExplicitVersion
                 ? this.ceProps<string>(`formatter.${formatterName}.explicitVersion`)
                 : match
-                    ? match[0]
-                    : result.stdout;
+                ? match[0]
+                : result.stdout;
```
Comment (jeremy-rifkin) [inline lib/handlers/formatting.ts:68]: This is one of the indent edge cases that prettier and eslint disagree on

**Pull request merged**